### PR TITLE
CNS-288 create basic pairing of project providers

### PR DESCRIPTION
**Pull request by Yaroms**

### CODE REVIEW

File: x/pairing/keeper/pairing_subscription_test.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	}{
+		{"happyflow", ts.spec.Apis[0].ComputeUnits, true},
+		{"epochCULimit", ts.plan.ComputeUnitsPerEpoch, false},
+	}
+
+	for i, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			relayRequest := &types.RelayRequest{
+				Provider:        ts.providers[0].address.String(),
```
Comment (omerlavanet) [inline x/pairing/keeper/pairing_subscription_test.go:85]: this is incompatible with how we build relays after the change, and creating the object with the keys like that leaves a lof ot lines to be changed, and i am tired already of manually changing how it's done with copy paste in all of our unitests, use a common function to build stub relays please

File: x/projects/keeper/project.go [hunk 2]
```diff
@@ -73,25 +73,26 @@ func (k Keeper) AddKeysToProject(ctx sdk.Context, projectID string, adminKey str
...
 
-	for _, chain := range project.Policy.ChainPolicies {
-		if chain.ChainId == chainID {
-			for _, api := range chain.Apis {
-				if api == apiName {
-					return true, project.Policy, nil
-				}
-			}
-		}
+	if project.Policy.TotalCuLimit <= project.UsedCu {
```
Comment (omerlavanet) [inline x/projects/keeper/project.go:91]: use a method

**Review by omerlavanet — CHANGES_REQUESTED** [review]
NICE JOB!

File: x/pairing/keeper/msg_server_relay_payment_test.go
[the remark this answers is not in the document]
Comment (omerlavanet) [reply x/pairing/keeper/msg_server_relay_payment_test.go:1]: created a task for it

**Review by omerlavanet — APPROVED** [review]

**Review by omerlavanet — APPROVED** [review]

File: proto/pairing/query.proto [hunk 3]
```diff
@@ -133,6 +133,8 @@ message QueryVerifyPairingRequest {
 message QueryVerifyPairingResponse {
   bool valid = 1;
   int64 index = 2;
+  uint64 pairedProviders = 3;
```
Comment (orenl-lava) [inline proto/pairing/query.proto:136]: nit: isn't the coding style in *.proto files like "paired_providers"?
and also "cu_per_epoch" below?

File: proto/projects/project.proto [hunk 4]
```diff
@@ -17,28 +17,30 @@ message Project {
...
 
 message ChainPolicy {
     string chain_id = 1;
     repeated string apis = 2 [(gogoproto.nullable) = false];
 }
 
-message ProtoString {
-    string string = 1;
+message ProtoDeveloperData {
+    string projectID = 1;
```
[the remark this answers is not in the document]
Comment (omerlavanet) [reply proto/projects/project.proto:44]: we need to align all ID, let's do it in a separate PR

**Review by omerlavanet — APPROVED** [review]

**Pull request merged**